lib/trie/proof.go

```diff
@@ -28,117 +27,41 @@ import (
 var (
 	// ErrEmptyTrieRoot occurs when trying to craft a prove with an empty trie root
 	ErrEmptyTrieRoot = errors.New("provided trie must have a root")
-
-	// ErrEmptyNibbles occurs when trying to prove or valid a proof to an empty key
-	ErrEmptyNibbles = errors.New("empty nibbles provided from key")
 )
 
-// GenerateProof constructs the merkle-proof for key. The result contains all encoded nodes
-// on the path to the key. Returns the amount of nodes of the path and error if could not found the key
-func (t *Trie) GenerateProof(key []byte, db chaindb.Writer) (int, error) {
-	key = keyToNibbles(key)
-	if len(key) == 0 {
-		return 0, ErrEmptyNibbles
-	}
-
-	var nodes []node
-	currNode := t.root
-
-proveLoop:
-	for {
-		switch n := currNode.(type) {
-		case nil:
-			return 0, errors.New("no more paths to follow")
+// GenerateProof receive the keys to proof, the trie root and a reference to database
+// will
+func GenerateProof(root []byte, keys [][]byte, db chaindb.Database) ([][]byte, error) {
+	trackedProofs := make(map[string][]byte)
 
-		case *leaf:
-			nodes = append(nodes, n)
+	for _, k := range keys {
+		nk := keyToNibbles(k)
 
-			if bytes.Equal(n.key, key) {
-				break proveLoop
-			}
-
-			return 0, errors.New("leaf node doest not match the key")
-
-		case *branch:
-			nodes = append(nodes, n)
-			if bytes.Equal(n.key, key) || len(key) == 0 {
-				break proveLoop
-			}
+		lookup := NewLookup(root, db)
+		recorder := new(Recorder)
 
-			length := lenCommonPrefix(n.key, key)
-			currNode = n.children[key[length]]
-			key = key[length+1:]
+		_, err := lookup.Find(nk, recorder)
+		if err != nil {
+			return nil, err
 		}
-	}
 
-	for _, n := range nodes {
-		var (
-			hashNode    []byte
-			encHashNode []byte
-			err         error
-		)
-
-		if encHashNode, hashNode, err = n.encodeAndHash(); err != nil {
-			return 0, fmt.Errorf("problems while encoding and hashing the node: %w", err)
-		}
+		fmt.Println("Len of records ", len(*recorder))
```

Comment:
I think should remove comment or make a log

Reply:
removed prints!